config/internal/configsource/manager.go

```diff
@@ -305,6 +313,109 @@ func (m *Manager) expandConfigSources(ctx context.Context, s string) (interface{
 	return retrieved.Value(), nil
 }
 
+// expandString expands environment variables and config sources that are specified on the string.
+func (m *Manager) expandString(ctx context.Context, s string) (interface{}, error) {
+	// Code based on os.Expand function. All delimiters that are checked against are
+	// ASCII so bytes are fine for this operation.
+	var buf []byte
+
+	// Using i, j, and w variables to keep correspondence with os.Expand code.
+	// i tracks the index in s from which a slice to be appended to buf should start.
+	// j tracks the char being currently checked and also the end of the slice to be appended to buf.
+	// w tracks the number of characters being consumed after a prefix identifying env vars or config sources.
+	i := 0
+	for j := 0; j < len(s); j++ {
+		if s[j] == expandPrefixChar && j+1 < len(s) {
+			if buf == nil {
+				buf = make([]byte, 0, 2*len(s))
+			}
+			buf = append(buf, s[i:j]...)
+
+			var retrieved interface{}
+			var err error
+			w := 0 // number of bytes consumed on this pass
+
+			switch {
+			case s[j+1] == expandPrefixChar:
+				// Escaped prefix:
+				buf = append(buf, s[j])
+				w = 1 // consumed a single char
+
+			case s[j+1] == '{':
+				// Bracketed usage, consume everything until first '}' exactly as os.Expand.
+				var content string
+				content, w = getShellName(s[j+1:])
+				content = strings.Trim(content, " ") // Allow for some spaces.
+				if len(content) > 1 && content[0] == expandPrefixChar {
+					name, _ := getShellName(content[1:])
```

Comment:
I am not sure what case this is. Is this something like `${$...}`?

Reply:
Yes, this proposal keeps it as `${ $file:/path/to/file }` syntax but easily changeable to ``${ file:/path/to/file }` if doubling the prefix looks too cumbersome.

Reply:
Removed the need for the internal `$`.